File: assign2/ruby/src/factorial.rb
     def recursion(num)
        raise illegalVaribale if num < 0
                
        if num <= 1
            1
        else
            num * recursion(num - 1)
        end
        
    end
    
    def tailRecursion(num, accumulator = 1)
        raise illegalVaribale if num < 0

        if num <= 1
            accumulator
        else
            tailRecursion(num - 1, num * accumulator)
        end
    end

File: assign2/elixir/lib/factorial.ex
defmodule Factorial do
    
	 def recursion(num) do
		cond do
            num < 0 ->
                 throw("IllegalArgument") 
	        num == 0 ->
                 1
            true ->
                 num * recursion(num - 1)    
			end
	 end


     def tailRecursion(num) do
        cond do
            num < 0 ->
                 throw("IllegalArgument")
            true ->
                 tailRecursion(num, 1)
        end
     end

     def tailRecursion(0, accumulator) do
        accumulator
     end

     def tailRecursion(num, accumulator) do
        tailRecursion(num - 1, accumulator * num)
     end
    
end

